test/unit/Governor.js

```diff
@@ -0,0 +1,558 @@
+import Fixture from "./helpers/Fixture"
+
+import truffleAssert from "truffle-assertions"
+import {utils, constants} from "ethers"
+
+const Governor = artifacts.require("Governor")
+const SetUint256 = artifacts.require("SetUint256")
+
+contract("Governor", accounts => {
+    let fixture
+    let governor
+    let setUint256
+
+    before(() => {
+        fixture = new Fixture(web3)
+    })
+
+    beforeEach(async () => {
+        await fixture.setUp()
+        governor = await Governor.new()
+        setUint256 = await SetUint256.new()
+    })
+
+    afterEach(async () => {
+        await fixture.tearDown()
+    })
+
+
+    const setUint256Tx = async (i, sender) => {
+        const txData = utils.arrayify(setUint256.contract.methods.setUint256(utils.bigNumberify(i)).encodeABI())
+        const dataHash = utils.solidityKeccak256(["address", "uint256", "bytes"], [setUint256.address, utils.bigNumberify("0"), txData])
+        const sig = await web3.eth.sign(dataHash, sender)
+        return {data: utils.hexlify(txData), hash: dataHash, sig: sig}
+    }
+
+    describe("constructor", () => {
+        it("initializes state: owner", async () => {
+            assert.equal(await governor.owner(), accounts[0])
+        })
+    })
+
+    describe("transferOwnership", () => {
+        it("reverts if not called by the contract itself, even if msg.sender is the owner", async () => {
+            await truffleAssert.reverts(
+                governor.transferOwnership(accounts[1]),
+                "unauthorized"
+            )
+        })
+
+        it("reverts if the new owner address is the zero-value for the address type", async () => {
+            const txData = utils.arrayify(governor.contract.methods.transferOwnership(constants.AddressZero).encodeABI())
+            await governor.stage(
+                {
+                    target: [governor.address],
+                    value: ["0"],
+                    data: [txData]
+                },
+                "0"
+            )
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [governor.address],
+                        value: ["0"],
+                        data: [txData]
+                    }
+                ),
+                "newOwner is zero"
+            )
+        })
+
+        it("updates ownership to a new owner", async () => {
```

Comment:
Let's also check that the contract storage value for `owner` is updated correctly?

Reply:
Bump

Reply:
fixed in 9c11912